client/utils.c

```diff
@@ -334,7 +334,7 @@ TDNFTouchFile(
     }
 
     old_mask = umask(022);
-    fd = creat(pszFile, S_IRUSR | S_IRGRP | S_IROTH);
+    fd = creat(pszFile, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
```

Comment:
I think adding this in %post of tdnf spec will be a good add-on
```
%post
test -d %{_var}/cache/%{name} && chmod -R 644 %{_var}/cache/%{name} || :
```
But assuming that /var/cache/tdnf is the configured repo dir (which holds good in most of the cases)

Reply:
It shouldn't be needed. When running as non-root, removing the file fails when it is read-only. But when run as root, it can be removed even when read-only. Since the system `tdnf` runs as root, there is no need to change permissions. `makecache/clean` wasn't working before the PR for non-root users, and the package is unaware of any non-root uses anyway.

Reply:
Agree. I checked in fedora, it allows regular user to do dnf makcache. Should we be in alignment?

Reply:
Thanks, I checked. I wasn't aware. `dnf` allows running `makecache` as an unprivileged user, but it stores the cache in `/var/tmp/dnf-<user>-<random uid/` (like `/var/tmp/dnf-okurth-50ituhym/`). Not in the system wide cache - which makes sense, we don't want a regular to have write access to that.

We could consider doing something similar, in another change. This PR would be a prerequisite for that.